pages/📊 Stable curve.py

```diff
@@ -16,147 +16,156 @@
-linear_invariant = LinearInvariant(x=base_x, y=base_y)
-
-df = pd.DataFrame()
-
-x_data_name = st.session_state["x_data"]["name"]
-df[x_data_name] = np.linspace(linear_invariant.constant*0.2, linear_invariant.constant*0.8, num=100)
-df["Current curve"] = current_stable_swape.calculate_y(df[x_data_name])
-df["New curve"] = new_stable_swape.calculate_y(df[x_data_name])
-
-df["Current spot price"] = current_stable_swape.calculate_spot_price(df[x_data_name])
-df["New spot price"] = new_stable_swape.calculate_spot_price(df[x_data_name])
-
-pool_fig = go.Figure(go.Scatter(
-    name = "Default Amp Factor",
-    x = df[x_data_name],
-    y = df["Current curve"],
-    hovertemplate =
-    '<b>Price: %{text}</b>',
-    text = df["Current spot price"].map('{:.5f}'.format)))
-
-pool_fig.add_trace(go.Scatter(
-    name = "New Amp Factor",
-    x = df[x_data_name],
-    y = df["New curve"],
-    hovertemplate = '<b>Price: %{text}</b>',
-    text = df["New spot price"].map('{:.5f}'.format)))
-
-pool_fig.update_layout(hovermode='x unified')
-
-if type_token_sell:
-  tokens_data = new_stable_swape.define_binary_sell_buy(type_token_sell, st.session_state["x_data"], st.session_state["y_data"])
-  type_token_buy = tokens_data['type_token_buy']
-  initial_amount_sell = tokens_data['initial_amount_sell']
-  initial_amount_buy = tokens_data['initial_amount_buy']
-  amount_token_sell = st.sidebar.number_input(label='How much of token {} you want to sell?'.format(type_token_sell),value=400.00, step=0.1,max_value=float(initial_amount_buy), min_value=0.0)
+if response["pool"] is None:
+  Components.error_container('Pool not found error', f'The pool {pool_id} do not exist in the {network} network')
+
+elif response["pool"]["poolType"] != "Stable":
+  Components.error_container('Not stable pool error', f'The {response["pool"]["name"]} pool is not stable')
+
+else:
+  st_utils = Streamlit()
+  st_utils.initiate_session_state("pool_data", response["pool"])
+  st_utils.initiate_session_state("x_data", st.session_state["pool_data"]["tokens"][0])
+  st_utils.initiate_session_state("y_data", st.session_state["pool_data"]["tokens"][1])
+  st_utils.initiate_session_state("names", [token["symbol"] for token in st.session_state["pool_data"]["tokens"]])
+  st_utils.initiate_session_state("balances", [float(token["balance"]) for token in st.session_state["pool_data"]["tokens"]])
+  st_utils.initiate_session_state("pool", {"pool_id": pool_id, "network": network})
+
+  st.title(f'{st.session_state["pool_data"]["name"]} Simulation')
+  col1, col2 = st.columns(2)
+
+  base_x = float(st.session_state["x_data"]["balance"])
+  base_y = float(st.session_state["y_data"]["balance"])
+  base_amp = float(st.session_state["pool_data"]["amp"])
+  amp_series = base_amp*np.append(np.logspace(-3, 0, endpoint=False), np.logspace(0, 3))
+
+  balance = px.pie(values=st.session_state["balances"], names=st.session_state["names"], labels=st.session_state["names"], height=350)
```

Comment:
On smaller screens this component is very strange to see. I'll commit a change that changes the legend placement to an absolute position